networking/types/types.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2016 The rkt Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package types
+
+import (
+	"fmt"
+	"net"
+
+	cnitypes "github.com/appc/cni/pkg/types"
+	"github.com/coreos/rkt/networking/netinfo"
+	"github.com/vishvananda/netlink"
+)
+
+// NetConf local struct extends cnitypes.NetConf with information about masquerading
+// similar to CNI plugins
+type NetConf struct {
+	cnitypes.NetConf
+	IPMasq bool `json:"ipMasq"`
+	MTU    int  `json:"mtu"`
+}
+
+type ActiveNet struct {
+	ConfBytes []byte
+	Conf      *NetConf
+	Runtime   *netinfo.NetInfo
+}
+
+// The following methods implement behavior of netDescriber by ActiveNet
```

Comment:
Yeah, the netDescriber interface. I wanted to add the `var _ netDescriber = blah blah` line, but that interface is in the package somewhere below stage1/init or somewhere. So I can't do it here, because that would likely end up with having circular imports. Also, networking module should not care about stage1 code.

This thing is a kind of hack, because IIRC most of the kvm specific code in networking package should actually be in CNI, not here. Also, for ActiveNet to implement some interface, you have to add the implementation in the same package as ActiveNet itself, right? Unless you do some more tricks. So all this is a kind of ew.